### ISSUE TITLE
Investigate whether a model trained on GPU can be loaded on CPU and vice versa, with pickle and state_dict. If that does not work, make it work.

### ISSUE THREAD

**Issue by benjamin-work** [dnouri/inferno#70]

**Comment by benjamin-work** [comment]
It seems that at least with pickle, it doesn't work:

```
...
  File "/opt/conda/envs/blotto/lib/python3.6/site-packages/torch/cuda/__init__.py", line 266, in _lazy_new
    _lazy_init()
  File "/opt/conda/envs/blotto/lib/python3.6/site-packages/torch/cuda/__init__.py", line 84, in _lazy_init
    _check_driver()
  File "/opt/conda/envs/blotto/lib/python3.6/site-packages/torch/cuda/__init__.py", line 58, in _check_driver
    http://www.nvidia.com/Download/index.aspx""")
AssertionError: 
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx

```

**Comment by ottonemo** [comment]
This should be fixed in #70.
